src/Apps/Artist/routes.tsx

```diff
@@ -87,42 +87,69 @@ export const routes: RouteConfig[] = [
     },
     query: graphql`
       query routes_ArtistTopLevelQuery($artistID: String!) @raw_response_type {
+        me {
+          id
+        }
         artist(id: $artistID) @principalField {
           ...ArtistApp_artist
           ...routes_Artist @relay(mask: false)
         }
       }
     `,
     render: ({ Component, props, match }) => {
-      if (Component && props) {
-        const { artist } = props as any
+      if (!(Component && props)) {
+        return null
+      }
 
-        if (!artist) {
-          return undefined
-        }
+      const { artist, me: user } = props as any
+      const { pathname } = match.location
 
-        const showArtistInsights =
-          showMarketInsights(artist) ||
-          (artist.insights && artist.insights.length > 0)
-        const hasArtistContent = hasOverviewContent(artist)
+      if (!artist) {
+        return undefined
+      }
 
-        const alreadyAtWorksForSalePath = match.location.pathname.includes(
-          `${artist.slug}/works-for-sale`
-        )
+      const showArtistInsights =
+        showMarketInsights(artist) ||
+        (artist.insights && artist.insights.length > 0)
+      const hasArtistContent = hasOverviewContent(artist)
 
-        const canShowOverview = showArtistInsights || hasArtistContent
+      const alreadyAtWorksForSalePath = pathname.includes(
+        `${artist.slug}/works-for-sale`
+      )
 
-        if (!canShowOverview && !alreadyAtWorksForSalePath) {
-          throw new RedirectException(`/artist/${artist.slug}/works-for-sale`)
-        }
+      const canShowOverview = showArtistInsights || hasArtistContent
+      /**
+       * The logic is as follows
+       *
+       * 1. If a user somehow ends up at /artist/<slug>/ redirect to /artist/<slug> because the former causes weird issues
+       * 2. If a user is logged in / redirects for /works-for-sale
+       * 3. If a user is logged in /overview opens the overview tab (which also links off to /overview)
+       * 4. If a user is not logged in / leads to the overview tab
+       * 5. If a user is not logged in /overview redirects to / and therefore the overview tab
+       * 6. If there's insufficient data, all tabs redirect to /works-for-sale
+       */
+      if (pathname === `/artist/${artist.slug}/`) {
```

Comment:
Interesting 😅

Reply:
Yeaaahh. It's almost considered a separate route, but it's not. You can navigate to it and it won't do the redirects and it also breaks the navigation tabs highlighting. 🤷‍♂